model/src/test/java/com/basistech/rosette/dm/AnnotatedTextTest.java

```diff
@@ -540,6 +540,9 @@ public void testBuildEntity() throws Exception {
         assertEquals(10, text.getEntityMentions().get(1).getStartOffset());
         assertEquals("PERSON", text.getEntityMentions().get(1).getEntityType());
         assertNull(text.getResolvedEntities());
+
+        // This would crash before the fix to ROS-226.
+        new Entity.Builder(text.getEntities().get(0));
```

Comment:
I wonder what we have to do to get an entity with null and not an empty
list for sentiment. Maybe it has to be read from json? I'll poke at the
test case more.

On Fri, Apr 29, 2016 at 7:54 AM, Joel Barry <notifications@github.com>
wrote:

> In model/src/test/java/com/basistech/rosette/dm/AnnotatedTextTest.java
> <https://github.com/basis-technology-corp/annotated-data-model/pull/5#discussion_r61565713>
> :
>
> > @@ -540,6 +540,9 @@ public void testBuildEntity() throws Exception {
> >          assertEquals(10, text.getEntityMentions().get(1).getStartOffset());
> >          assertEquals("PERSON", text.getEntityMentions().get(1).getEntityType());
> >          assertNull(text.getResolvedEntities());
> > +
> > +        // This would crash before the fix to ROS-226.
> > +        new Entity.Builder(text.getEntities().get(0));
>
> Reverting the fix above and running this test does not provoke a crash for
> me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/basis-technology-corp/annotated-data-model/pull/5/files/56c02abc49f85dbf680941499910df9639fdc779#r61565713>
>